data/tools/wesnoth/wmltools3.py

```diff
@@ -477,7 +499,7 @@ def __str__(self):
 
 class CrossRef:
     macro_reference = re.compile(r"\{([A-Z_][A-Za-z0-9_:]*)(?!\.)\b")
-    file_reference = re.compile(r"[A-Za-z0-9{}.][A-Za-z0-9_/+{}.@\-\[\],~\*]*(" + "|".join(resource_extensions) + ")(?=(~.*)?)")
+    file_reference = re.compile(r"([A-Za-z0-9{}.][A-Za-z0-9_/+{}.@\-\[\],~\*]*(" + "|".join(resource_extensions) + "))((~[A-Z]+\(.*\))*)(:([0-9]+|\[[0-9,*~]*\]))?")
```

Comment:
While testing the former regex, I noticed that it had issues with multiple filenames on the same line even if durations weren't specified. For this reason, I'd try making the second character set non greedy (`*?`) and adding a match to a literal dot after it (`\.`).

Reply:
I agree that the old regex had issues even without durations. I don't quite see what change you are suggesting or what difference it should make though.

Reply:
Ah, you mean adding a mandatory period in front of the extension? I suppose it would help.